code/game/objects/items/weapons/cosmetics.dm

```diff
@@ -82,3 +82,104 @@
 
 /obj/item/weapon/haircomb/attack_self(mob/user)
 	user.visible_message("<span class='notice'>[user] uses [src] to comb their hair with incredible style and sophistication. What a [user.gender == FEMALE ? "lady" : "guy"].</span>")
+
+/obj/item/weapon/razor
+	name = "electric razor"
+	desc = "The latest and greatest power razor born from the science of shaving."
+	icon = 'icons/obj/weapons.dmi'
+	icon_state = "razor"
+	w_class = 2
+
+/obj/item/weapon/razor/proc/shave(mob/living/carbon/human/H, location)
+	if(location == "head")
+		H.h_style = H.species.default_h_style
+	else
+		H.f_style = H.species.default_f_style
+
+	H.update_hair()
+	playsound(H, 'sound/items/welder2.ogg', 20, 1)
+
+
+/obj/item/weapon/razor/attack(mob/M, mob/user, var/target_zone)
+	if(!ishuman(M))
+		return
+
+	var/mob/living/carbon/human/H = M
+	var/obj/item/organ/external/E = H.get_organ(target_zone)
+
+	if(!E || E.is_stump())
+		to_chat(user, "<span class='danger'>They are missing that limb!</span>")
+		return FALSE
+
+	if(!ishuman_species(H) && !istajara(H))	//you can only shave humans and tajara for obvious reasons
+		return FALSE
+
+
+	if(target_zone == "head")
+		if(H.head && (H.head.body_parts_covered & HEAD))
+			to_chat(user, "<span class='warning'>\The [H.head] is in the way!</span>")
+			return FALSE
+
+		if(H.h_style == "Bald" || H.h_style == "Balding Hair" || H.h_style == "Skinhead" || H.h_style == "Tajaran Ears")
+			to_chat(user, "<span class='warning'>There is not enough hair left to shave!</span>")
+			return FALSE
+
+		if(H == user) //shaving yourself
+			user.visible_message("[user] starts to shave \his head with \the [src].", \
+									 "<span class='notice'>You start to shave your head with \the [src].</span>")
+			if(do_after(user, 20))
+				user.visible_message("[user] shaves \his head with \the [src].", \
+										 "<span class='notice'>You finish shaving with \the [src].</span>")
+				shave(H, target_zone)
+
+			return TRUE
+
+		else
+			user.visible_message("<span class='warning'>[user] tries to shave [H]'s head with \the [src]!</span>", \
+									 "<span class='notice'>You start shaving [H]'s head.</span>")
+			if(do_after(user, 20))

```

Comment:
You should never use do_after when you're doing something to another mob. Using just do_after will cause a bug that will allow you to shave a person's head even if they move across the room.

Syntax:
```/proc/do_mob(mob/user, mob/target, delay = 30, needhand = TRUE, display_progress = TRUE, datum/callback/extra_checks)```